docs/book/src/basics/commonly_used_library_types.md

```diff
@@ -0,0 +1,44 @@
+# Commonly Used Library Types
+
+The Sway Standard Library is the foundation of portable Sway software, a set of minimal shared abstractions for the broader Sway ecosystem. It offers core types, library-defined operations on language primitives, native asset management, blockchain contextual operations, access control, storage management, and support for types from other VMs, among many other things. Reference the standard library docs [here](https://fuellabs.github.io/sway/master/std/index.html).
+
+## `Result<T, E>`
+
+Type `Result` is the type used for returning and propagating errors. It is an `enum` with two variants: `Ok(T)`, representing success and containing a value, and `Err(E)`, representing error and containing an error value. The `T` and `E` in this definition are type parameters, allowing `Result` to be generic and to be used with any types.
+
+```sway
+pub enum Result<T, E> {
+    Ok: T,
+    Err: E,
+}
+```
```

Comment:
Could use a link internally to display the type (similar to just below) instead of hard coding into docs

Reply:
We have a [precedence](https://github.com/FuelLabs/sway/blob/91feedc9c75193c8415fb5f566929626a4479a96/sway-lib-std/src/identity.sw#L8) for using anchors in the standard library. We could do the same here.

Reply:
Done.